importers/research-media-csv-importer.php

```diff
@@ -0,0 +1,249 @@
+	class CSVThumbnailImporter {
+		private
+			$csv_data,
+			$base_url,
+			$source_meta_key,
+			$matched_records,
+			$media_base_url,
+			$force_update = false,
+
+			$processed            = 0,
+			$skipped_missing_info = 0,
+			$skipped_no_match     = 0,
+			$matched              = 0,
+			$thumbnails_exist     = 0,
+			$thumbnails_added     = 0,
+			$thumbnails_deleted   = 0,
+			$thumbnails_errors     = 0;
+
+		/**
+		 * Constructor
+		 * @author Jim Barnes
+		 * @since 2.0.0
+		 * @param string $filepath The path to the WordPress export XML file
+		 * @param string $base_url The URL to the WordPress API to retrieve files from
+		 */
+		public function __construct( $filepath, $base_url, $force_update = false ) {
+			$this->matched_records = array();
+			$this->force_update = $force_update;
+			$this->base_url = $base_url;
+
+			try {
+				$rows = array_map( 'str_getcsv', file( $filepath ) );
+				$headers = array_shift( $rows );
+				$this->csv_data = array();
+
+				foreach( $rows as $row ) {
+					$this->csv_data[] = array_combine( $headers, $row );
+				}
+			} catch (Exception $e) {
+				throw $e;
+			}
+
+			$this->media_base_url = "{$this->base_url}/wp-json/wp/v2/media";
+		}
+
+		/**
+		 * Attempts to match a person from the importer
+		 * using the provided email addresses.
+		 * @author Jim Barnes
+		 * @since 2.0.0
+		 * @param string $email The email address from the file
+		 * @return int The post ID of the returned record
+		 */
+		private function match_person_by_email( $email ) {
+			$args = array(
+				'post_type'      => 'person',
+				'posts_per_page' => 1,
+				'meta_query'     => array(
+					array(
+						'key'     => 'person_email',
+						'value'   => $email,
+						'compare' => '='
+					)
+				)
+			);
+
+			$posts = get_posts( $args );
+
+			if ( count( $posts ) > 0 ) {
+				return $posts[0]->ID;
+			}
+
+			return null;
+		}
+
+		/**
+		 * Helper function for retrieving JSON
+		 * values from the WP remote API.
+		 * @author Jim Barnes
+		 * @since 2.0.0
+		 * @param string $url The URL to fetch
+		 * @return mixed JSON-decoded object or false on failure
+		 */
+		private function fetch_json( $url ) {
+			$response      = wp_remote_get( $url, array( 'timeout' => 10 ) );
+			$response_code = wp_remote_retrieve_response_code( $response );
+			$result        = false;
+
+			if ( is_array( $response ) && is_int( $response_code ) && $response_code < 400 ) {
+				$result = json_decode( wp_remote_retrieve_body( $response ) );
+			}
+
+			return $result;
+		}
+
+		/**
+		 * Gets the image data for the person
+		 * @author Jim Barnes
+		 * @since 2.0.0
+		 * @param int $id The remote ID of the person
+		 * @return mixed
+		 */
+		private function get_person_image( $id ) {
+			$url = "{$this->media_base_url}/{$id}";
+			$media_data = $this->fetch_json( $url );
+
+			if ( $media_data ) {
+				$retval = $media_data->source_url;
+				if ( strpos( $retval, strval( $this->base_url ) ) == -1 ) {
```

Comment:
Strict equality operator would be ideal here.  I'm also not sure that `strpos()` can ever return `-1`?